### PR TITLE
DRPL-11: Add backend for cm-hero-teaser.

### PR DESCRIPTION
**Pull request by ChristianNiemitzRossant**
Add Hero-Teaser without demo content

### CODE REVIEW

**Comment by sharique** [comment]
I setup this in local in when importing default content I got following error
```
[error]  Drupal\Core\Database\IntegrityConstraintViolationException: SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'uid' cannot be null: INSERT INTO "media_field_data" ("mid", "vid", "bundle", "langcode", "status", "uid", "name", "thumbnail__target_id", "thumbnail__alt", "thumbnail__title", "thumbnail__width", "thumbnail__height", "created", "changed", "default_langcode", "revision_translation_affected") VALUES (:db_insert_placeholder_0, :db_insert_placeholder_1, :db_insert_placeholder_2, :db_insert_placeholder_3, :db_insert_placeholder_4, :db_insert_placeholder_5, :db_insert_placeholder_6, :db_insert_placeholder_7, :db_insert_placeholder_8, :db_insert_placeholder_9, :db_insert_placeholder_10, :db_insert_placeholder_11, :db_insert_placeholder_12, :db_insert_placeholder_13, :db_insert_placeholder_14, :db_insert_placeholder_15); Array
(
    [:db_insert_placeholder_0] => 1
    [:db_insert_placeholder_1] => 1
    [:db_insert_placeholder_2] => image
    [:db_insert_placeholder_3] => de
    [:db_insert_placeholder_4] => 1
    [:db_insert_placeholder_5] => 
    [:db_insert_placeholder_6] => 0_Startseite_1900x600_v2.jpg
    [:db_insert_placeholder_7] => 1
    [:db_insert_placeholder_8] => an Image
    [:db_insert_placeholder_9] => 
    [:db_insert_placeholder_10] => 1900
    [:db_insert_placeholder_11] => 600
    [:db_insert_placeholder_12] => 1647903320
    [:db_insert_placeholder_13] => 1647903320
    [:db_insert_placeholder_14] => 1
    [:db_insert_placeholder_15] => 1
)
 in Drupal\Core\Database\Driver\mysql\ExceptionHandler->handleExecutionException() (line 50 of /app/web/core/lib/Drupal/Core/Database/Driver/mysql/ExceptionHandler.php). 

```

**Comment by sharique** [comment]
If you are not able to fix it, remove default content and we will create is manually.

File: config/default/core.entity_view_display.paragraph.hero_teaser_item.default.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    settings:
+      link_to_entity: false
+    third_party_settings: {  }
+    weight: 1
+    region: content
+  field_media:
+    type: entity_reference_entity_view
+    label: hidden
+    settings:
+      view_mode: media_library
```
Comment (sharique) [inline config/default/core.entity_view_display.paragraph.hero_teaser_item.default.yml:31]: Create new display mode to default, will check during in front end implementation how to show based on field value.

File: config/default/core.entity_view_display.paragraph.hero_teaser_item.default.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    - field.field.paragraph.hero_teaser_item.field_text_align
+    - paragraphs.paragraphs_type.hero_teaser_item
+  module:
+    - text
+id: paragraph.hero_teaser_item.default
+targetEntityType: paragraph
+bundle: hero_teaser_item
+mode: default
+content:
+  field_headline_hti:
```
Comment (sharique) [inline config/default/core.entity_view_display.paragraph.hero_teaser_item.default.yml:19]: use existing headline field.

File: config/default/core.entity_view_display.paragraph.hero_teaser_item.default.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    region: content
+  field_teaser_text:
+    type: text_default
+    label: hidden
+    settings: {  }
+    third_party_settings: {  }
+    weight: 2
+    region: content
+hidden:
+  field_link_url: true
```
Comment (sharique) [inline config/default/core.entity_view_display.paragraph.hero_teaser_item.default.yml:44]: Change field name to field_link only.

File: config/default/field.field.paragraph.hero_teaser.field_display_indicators.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+description: "Defines if the Indicators to change the Slide are shown \r\nunchecked (Default) => No Indicators are shown \r\nChecked => Indicators are shown (Indicators need to be defined) "
+required: true
+translatable: false
+default_value:
+  -
+    value: 1
+default_value_callback: ''
+settings:
+  on_label: 'On'
+  off_label: 'Off'
```
Comment (sharique) [inline config/default/field.field.paragraph.hero_teaser.field_display_indicators.yml:22]: Change labels to Yes and No.

File: config/default/field.field.paragraph.hero_teaser_item.field_link_url.yml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  config:
+    - field.storage.paragraph.field_link_url
+    - paragraphs.paragraphs_type.hero_teaser_item
+  module:
+    - link
+id: paragraph.hero_teaser_item.field_link_url
+field_name: field_link_url
+entity_type: paragraph
+bundle: hero_teaser_item
+label: 'Link URL'
```
Comment (sharique) [inline config/default/field.field.paragraph.hero_teaser_item.field_link_url.yml:14]: Change label to `Link`

**Review by sharique — CHANGES_REQUESTED** [review]

File: config/default/core.entity_view_display.paragraph.hero_teaser.default.yml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  module:
+    - entity_reference_revisions
+id: paragraph.hero_teaser.default
+targetEntityType: paragraph
+bundle: hero_teaser
+mode: default
+content:
+  field_hero_teaser_item:
+    type: entity_reference_revisions_entity_view
+    label: above
```
Comment (sharique) [inline config/default/core.entity_view_display.paragraph.hero_teaser.default.yml:20]: Hide label.

File: config/default/field.storage.paragraph.field_headline_hti.yml [hunk 7]
```diff
@@ -0,0 +1,21 @@
+uuid: 014dcf10-aca5-4286-85fb-81af267b14e5
+langcode: de
+status: true
+dependencies:
+  module:
+    - paragraphs
+id: paragraph.field_headline_hti
```
Comment (sharique) [inline config/default/field.storage.paragraph.field_headline_hti.yml:7]: Use existing headline field.

File: config/default/field.storage.paragraph.field_text_align.yml [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    - paragraphs
+id: paragraph.field_text_align
+field_name: field_text_align
+entity_type: paragraph
+type: list_string
+settings:
+  allowed_values:
+    -
+      value: left
+      label: left
```
Comment (sharique) [inline config/default/field.storage.paragraph.field_text_align.yml:16]: Start label with capital letters.

**Review by sharique — CHANGES_REQUESTED** [review]

File: composer.json [hunk 9]
```diff
@@ -71,6 +71,12 @@
         "composer-exit-on-patch-failure": true,
         "patchLevel": {
             "drupal/core": "-p2"
+        },
+        "enable-patching": "true",
+        "patches": {
+            "drupal/default_content_deploy": {
+                "Avoid uid = 0 for import": "https://www.drupal.org/files/issues/2022-01-21/default_content_deploy-entity-uid-3130782-7.patch"
+            }
```
Comment (sharique) [inline composer.json:79]: @ChristianNiemitzRossant you also need to commit updated composer.lock after applying patch.  There might be some conflict with composer.lock, as I updated drupal core.

Comment (ChristianNiemitzRossant) [reply composer.json:79]: I hopefully solved it.

**Comment by ChristianNiemitzRossant** [comment]
Unconflicted.

**Comment by sharique** [comment]
merge but few small changes can be added in next PR.

- Use default display mode for media display.
- For Media hide media name.
- In default content add one video also.

**Pull request merged**